File: go/575.go
func distributeCandies(candyType []int) int {
    set := make(map[int]bool)
    for _,candy := range candyType {
        set[candy] = true
    }

    unqLen := len(set)
    ogLen := len(candyType)
    r := 0

    for r < unqLen && r != ogLen/2 {
        r++
    }

    return r
}

File: py/575.py
class Solution:
    def distributeCandies(self, candyType:List[int]) -> int:
        unq = len(set(candyType))
        candyLen = len(candyType)
        r = 0
        while r < unq and r != candyLen/2:
            r += 1
        return r
